// fastify-type-provider-zod/dist/src/errors.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createValidationError = exports.InvalidSchemaError = exports.ResponseSerializationError = void 0;
const error_1 = __importDefault(require("@fastify/error"));
class ResponseSerializationError extends (0, error_1.default)('FST_ERR_RESPONSE_SERIALIZATION', "Response doesn't match the schema", 500) {
    constructor(cause, method, url) {
        super({ cause });
        this.cause = cause;
        this.method = method;
        this.url = url;
    }
}
exports.ResponseSerializationError = ResponseSerializationError;
exports.InvalidSchemaError = (0, error_1.default)('FST_ERR_INVALID_SCHEMA', 'Invalid schema passed: %s', 500);
const createValidationError = (error, method, url) => error.errors.map((issue) => ({
    keyword: issue.code,
    instancePath: `/${issue.path.join('/')}`,
    schemaPath: `#/${issue.path.join('/')}/${issue.code}`,
    params: {
        issue,
        zodError: error,
        method,
        url,
    },
    message: error.message,
}));
exports.createValidationError = createValidationError;
